connect-inject/container_init.go

```diff
@@ -323,6 +323,21 @@ func (h *Handler) containerInit(pod *corev1.Pod, k8sNamespace string) (corev1.Co
 // and the connect-proxy service should come after the "main" service
 // because its alias health check depends on the main service to exist.
 const initContainerCommandTpl = `
+{{- if .AuthMethod }}
+consul-k8s consul-init -method="{{ .AuthMethod }}" \
+  {{- if.ConsulNamespace }}
+  {{- if .NamespaceMirroringEnabled }}
+  {{- /* If namespace mirroring is enabled, the auth method is
+         defined in the default namespace */}}
+  -namespace="default" \
+  {{- else }}
+  -namespace="{{ .ConsulNamespace }}" \
+  {{- end }}
+  {{- end }}
+  -meta="pod=${POD_NAMESPACE}/${POD_NAME}"
+{{- /* The acl token file needs to be read by the consul-sidecar which runs
+       as non-root user consul-k8s. TODO: will this be necessary anymore? */}}
+{{- end }}
```

Comment:
When the endpoints controller is ready we will remove everything past this in the command tmpl